stacks-signer/src/config.rs

```diff
@@ -38,25 +43,68 @@ pub enum ConfigError {
     BadField(String, String),
 }
 
+#[derive(serde::Deserialize, Debug, Clone)]
+#[serde(rename_all = "lowercase")]
+/// The Stacks network to use.
+pub enum Network {
+    /// The mainnet network
+    Mainnet,
+    /// The testnet network
+    Testnet,
+}
+
 /// The parsed configuration for the signer
 pub struct Config {
     /// endpoint to the stacks node
     pub node_host: SocketAddr,
+    /// endpoint to the stackerdb receiver
+    pub endpoint: SocketAddr,
     /// smart contract that controls the target stackerdb
     pub stackerdb_contract_id: QualifiedContractIdentifier,
-    /// the private key used to sign blocks, chunks, and transactions
-    pub private_key: StacksPrivateKey,
+    /// The Scalar representation of the private key for signer communication
+    pub message_private_key: Scalar,
+    /// The signer's Stacks private key
+    pub stacks_private_key: StacksPrivateKey,
+    /// The network to use. One of "mainnet" or "testnet".
+    pub network: Network,
+    /// The signer ID and key ids mapped to a pulbic key
+    pub signer_ids_public_keys: PublicKeys,
+    /// The signer IDs mapped to their Key IDs
+    pub signer_key_ids: SignerKeyIds,
+    /// This signer's ID
+    pub signer_id: u32,
+    /// The time to wait for a response from the stacker-db instance
+    pub event_timeout: Duration,
+}
+
+/// Internal struct for loading up the config file signer data
+#[derive(Clone, Deserialize, Default, Debug)]
+struct RawSigners {
+    pub public_key: String,
+    pub key_ids: Vec<u32>,
 }
 
 /// Internal struct for loading up the config file
 #[derive(Deserialize, Debug)]
 struct RawConfigFile {
     /// endpoint to stacks node
     pub node_host: String,
+    /// endpoint to stackerdb receiver
+    pub endpoint: String,
     /// contract identifier
     pub stackerdb_contract_id: String,
-    /// the private key used to sign blocks, chunks, and transactions in hexademical format
-    pub private_key: String,
+    /// the 32 byte ECDSA private key used to sign blocks, chunks, and transactions
+    pub message_private_key: String,
+    /// The hex representation of the signer's Stacks private key
+    pub stacks_private_key: String,
```

Comment:
I'm having a hard time understanding the difference between `message_private_key` and `stacks_private_key`.  Can you help me understand why there need to be two private keys?

Reply:
I can update the comment on the stacks_private_key as well to indicate what it is used for if that helps (communicating with the stacks node/stacker db contract)?

Reply:
Thanks for updating the comment @jferrant!

@xoloki I agree in principal that the keys ought to be separate and have different storage strategies, but I'm curious why the signer needs to have a key that controls large amounts of STX in the first place?  When the user stacks their STX in `pox-4`, they'd additionally set a dedicated signing key for the purposes of signing blocks, chunks, and transactions (i.e. the counterpart to `message_private_key`).  The STX would presumably be managed by the user's wallet, not the signer.  Additionally, the PoX rewards would be sent to a (separate) address of the user's choosing, which almost certainly would not be under the control of the signer.  Please help me understand what I'm missing?

Reply:
Totally possible that I'm making bad assumptions here.  I though the way PoX worked was that users stacked significant (~100k) STX for each reward slot, and that the STX address associated with the stacking wallets would be the same STX address that Signers will use to talk to sbtc contracts etc.

If this is a bad assumption, and the STX address the Signer will use with stackerDB is not the address which controls the large STX stake, then yeah we can just use the same private key for both.

Reply:
The users don't stack to specific reward slots per se; they just stack a large quantity of STX, and depending on how much STX they stacked relative to everyone else, they are allotted zero or more reward slots by the system at the start of the next reward cycle.

That said, the key that holds the STX and performs the stacking operation would _not_ need to be known to the signer.  Instead, the act of stacking from this high-value key would assign a _different_ block-signing key for the signer to use.  Depending on what you think about this, it may even be desirable to have the user delegate _two_ keys to the signer: one for signing Stacks blocks, and one for signing Bitcoin sBTC transactions.  But in no case would the signer ever possess the key that controls the user's STX.